### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by sphen13**
OK - went through some last changes as we discussed.  Just reviewed all the changes - I think this is all good - but please review to make sure I didnt miss anything.

My hope is this is a good staring point and I can make the docker container more mature as time goes on.

### CODE REVIEW

**Comment by sphen13** [comment]
Just updated with some bugs that @kginger and I found.  Please test against the latest here.

File: Server/emailHelper-dist.sh [hunk 1]
```diff
@@ -29,6 +29,12 @@ smtpServer=""
 smtpAuth=""
 smtpPass=""
 
+if [[ ${IN_DOCKER} && ${SMTP_SERVER} && ${SMTP_AUTH} && ${SMTP_PASS} ]]; then
+	smtpServer=$SMTP_SERVER
```
Comment (bestmacs) [inline Server/emailHelper-dist.sh]: Walking through the docker setup in the readme, I did not see where to provide these.

File: Server/jail.local [hunk 2]
```diff
@@ -241,12 +241,12 @@ maxretry = 2
 
 # http://linuxaria.com/howto/how-to-protect-apache-with-fail2ban
 
-[apache-phpmyadmin]
-enabled  = true
-port     = http,https
-filter   = apache-phpmyadmin
-logpath  = /var/log/apache*/*error.log
-maxretry = 3
+#[apache-phpmyadmin]
```
Comment (bestmacs) [inline Server/jail.local]: Why comment these out?

File: docker/DOCKER_FULL_EXAMPLE.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+**Wait a minute or two for the MySQL container to initialize.** You can get the status of the container by running `docker ps -a`.  Wait until you see it with a status of **(healthy)**
+
+#### Create BlueSky container
+
+> **Note:** _All the variables that need to change.  If you use complex passwords with offending characters you should enclose the password in single quotes.  Passwords do not work with a backslash in them_
+
+> **Note:** _We do not care about SSL certs here as the Caddy container will take care of that_
+
+```
+docker run -d --name bluesky \
```
Comment (bestmacs) [inline docker/DOCKER_FULL_EXAMPLE.md]: Ah, here they are.  I missed these on the first trial run.

File: Server/emailHelper-dist.sh (hunk 1 above)
Comment (sphen13) [reply Server/emailHelper-dist.sh]: assume this is one of the comments you figured out later - in the docker/README.md (https://github.com/logicnow/BlueSky/pull/14/files#diff-a489144702a723431811d56a3c44690a) it lists these 3 SMTP ENV variables :)

File: Server/jail.local (hunk 2 above)
Comment (sphen13) [reply Server/jail.local]: you are correct - i dont need to comment out - i may have gotten a bit over zealous.  it just assumes that the user is using phpmyadmin which they may not (not part of the bluesky setup or instructions).  i can uncomment in this PR and get this jail back to what it was.

File: docker/DOCKER_FULL_EXAMPLE.md (hunk 3 above)
Comment (sphen13) [reply docker/DOCKER_FULL_EXAMPLE.md]: yes so in this first pass (i can do more in the future) i just focussed on adapting what you did for docker.  that includes not really worrying about SSL much (no auto LetsEncrypt).  I decided in the "full setup" example to walk the user through creating a caddy container which would provide SSL automatically for the bluesky container.  in the case that the user did not want to do this and wanted to use their own certs i have a way to pass the cert and private key to the container which is have tested and works.

**Comment by sphen13** [comment]
undid the jail.local comment block - not only necessary changes have been made

**Comment by sphen13** [comment]
woah

**Pull request merged**